plasmapy/simulation/tests/test_particle_tracker.py

```diff
@@ -0,0 +1,137 @@
+"""
+Tests for particle_tracker.py
+"""
+import astropy.units as u
+import numpy as np
+import pytest
+
+from hypothesis import given, settings
+from hypothesis import strategies as st
+from hypothesis.extra.numpy import arrays
+
+from plasmapy.formulary.lengths import gyroradius
+from plasmapy.particles import CustomParticle
+from plasmapy.plasma.grids import CartesianGrid
+from plasmapy.simulation.particle_tracker import (
+    MemoryIntervalSaveRoutine,
+    NoFieldsStoppingCondition,
+    ParticleTracker,
+    TimeElapsedStopCondition,
+)
+
+
+class TestParticleTrackerGyroradius:
+    point_particle = CustomParticle(1 * u.kg, 1 * u.C)
+
+    def instantiate_simulation(self, B_strength=1 * u.T):
+        L = 1e6 * u.km
+        num = 2
+        grid = CartesianGrid(-L, L, num=num)
+        grid_shape = (num,) * 3
+
+        Bz = np.full(grid_shape, B_strength) * u.T
+        grid.add_quantities(B_z=Bz)
+
+        return ParticleTracker(
+            grid, req_quantities=["E_x", "E_y", "E_z", "B_x", "B_y", "B_z"]
+        )
+
+    @pytest.mark.slow()
```

Comment:
Just curious: is this actually slow? I'd hope it would be pretty quick for just a couple particles. You only need ~50-100 iterations/gyroperiod and ~10 gyroperiods, so you shouldn't need more than 1000 iterations.

Reply:
It seems to do fine with a step of dt=1e-2, the resulting relative differences are usually ~5e-2, so these are the parameters I'll use for the assertion. Good catch!